cloudnet/src/main/java/de/dytanic/cloudnet/command/commands/CommandModules.java

```diff
@@ -20,82 +22,179 @@ public CommandModules() {
+            }
+
+        } else if (args.length == 3 && args[0].equalsIgnoreCase("install")) {
+
+            ModuleId moduleId = new ModuleId(args[1], args[2]);
+            RepositoryModuleInfo moduleInfo = moduleRepository.loadRepositoryModuleInfo(moduleId);
+
+            if (moduleInfo == null) {
+                sender.sendMessage(LanguageManager.getMessage("command-modules-repository-module-not-found").replace("%id%", moduleId.toString()));
+                return;
+            }
+
+            ModuleInstaller moduleInstaller = super.getCloudNet().getModuleInstaller();
+
+            if (moduleInstaller.isModuleInstalled(moduleId)) {
+                sender.sendMessage(LanguageManager.getMessage("command-modules-repository-module-already-installed").replace("%id%", moduleId.toString()));
+                return;
+            }
+
+            try {
+                moduleInstaller.installModule(super.getCloudNet().getConsole(), moduleRepository.getBaseURL(), moduleInfo);
 
-                this.displayModuleInfo(sender, wrapper);
+                sender.sendMessage(LanguageManager.getMessage("command-modules-install-success").replace("%id%", moduleId.toString()));
+            } catch (IOException exception) {
+                exception.printStackTrace();
             }
+
+        } else if (args.length == 3 && args[0].equalsIgnoreCase("uninstall")) {
+
+            ModuleId moduleId = new ModuleId(args[1], args[2]);
+
+            ModuleInstaller moduleInstaller = super.getCloudNet().getModuleInstaller();
+
+            if (!moduleInstaller.isModuleInstalled(moduleId)) {
+                sender.sendMessage(LanguageManager.getMessage("command-modules-repository-module-not-installed").replace("%id%", moduleId.toString()));
+                return;
+            }
+
+            try {
+                moduleInstaller.uninstallModule(moduleId);
+
+                sender.sendMessage(LanguageManager.getMessage("command-modules-uninstall-success").replace("%id%", moduleId.toString()));
+            } catch (IOException exception) {
+                exception.printStackTrace();
+            }
+
         }
+    }
 
-        IModuleWrapper wrapper = CloudNetDriver.getInstance().getModuleProvider().getModule(args[0]);
-        if (wrapper != null) {
-            this.displayModuleInfo(sender, wrapper);
+    private boolean matchesProperties(ModuleId moduleId, Properties properties) {
+        if (properties.containsKey("group") && !moduleId.getGroup().contains(properties.get("group"))) {
+            return false;
+        }
+        if (properties.containsKey("name") && !moduleId.getName().contains(properties.get("name"))) {
+            return false;
         }
+        if (properties.containsKey("version") && !moduleId.getVersion().contains(properties.get("version"))) {
+            return false;
+        }
+        return true;
+    }
+
+    private void displayRemoteModuleInfo(ICommandSender sender, RepositoryModuleInfo moduleInfo) {
```

Comment:
Codacy found an issue: [The method 'displayRemoteModuleInfo(ICommandSender, RepositoryModuleInfo)' has an NPath complexity of 2187](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=5772469)